common/Makefile.am

```diff
@@ -49,3 +49,40 @@ noinst_HEADERS = \
 	pdnsexception.hh \
 	wforce-webserver.hh \
 	login_tuple.hh
+
+if UNIT_TESTS
+
+if HAVE_BOOST_GE_148
+TESTS_ENVIRONMENT = env BOOST_TEST_LOG_LEVEL=message SRCDIR='$(srcdir)'
+TESTS=testrunner
+noinst_PROGRAMS = \
+	testrunner
+else
+check-local:
+	@echo "Unit tests disabled, boost is too old"
+endif
+
+else
+check-local:
+	@echo "Unit tests are not enabled"
+	@echo "Run ./configure --enable-unit-tests"
+endif
+
+testrunner_SOURCES = \
+	testrunner.cc \
+	test-minicurl.cc
+
+testrunner_LDFLAGS = \
+	$(AM_LDFLAGS) \
+	-pthread \
```

Comment:
I *think* boost already adds `-pthread` flag.

Reply:
Also i think you should look at m4/pdns_check_os.m4

Reply:
Well no harm to do it twice

Reply:
except if you are not compiling on linux =)

Reply:
wforce is not designed to work on any OS that doesn't have pthreads

Reply:
except some os'es have extra things you need to do, and it's not always called -pthread. Some have -lpthread instead.

Reply:
Ah I see